SignalServiceKit/src/Account/TSAccountManager.h

```diff
@@ -116,6 +114,22 @@ extern NSString *const kNSNotificationName_LocalNumberDidChange;
 
 + (void)unregisterTextSecureWithSuccess:(void (^)(void))success failure:(void (^)(NSError *error))failureBlock;
 
+#pragma mark - De-Registration
+
+// De-registration reflects whether or not the client has received
+// a 403 from the service.
+- (BOOL)isDeregistered;
+- (void)setIsDeregistered:(BOOL)isDeregistered;
+
+#pragma mark - Re-registration
+
+// Re-registration is the process of re-registering _with the same phone number_.
+
+// Returns YES on success.
+- (BOOL)resetForReregistration;
+- (NSString *)reregisterationPhoneNumber;
```

Comment:
This isn't the same as `phoneNumberAwaitingVerification`.  'phoneNumberAwaitingVerification' isn't persisted, for one thing.

Reply:
Also, during re-registration (as opposed to initial registration) users are "locked" into their old phone number to avoid edge cases.